cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/search/AzureSearch.scala

```diff
@@ -142,7 +143,7 @@ class AddDocuments(override val uid: String) extends CognitiveServicesBase(uid)
   override def responseDataType: DataType = ASResponses.schema
 }
 
-object AzureSearchWriter extends IndexParser with SLogging {
+object AzureSearchWriter extends IndexParser with IndexJsonGetter with VectorColsParser with SLogging {
```

Comment:
I agree with your suggestion for VectorColParser. For IndexJsonGetter, I am thinking we could use this in unit tests to validate a column we create is indeed a vector column. So we might also use it in unit tests. Also another reason why I put it in AzureSearchAPI was because other calls like IndexLister is in there, so I thought what the function does fits the theme. What do you think?

Reply:
if you use it twice later on makes sense!